### PR TITLE
[CMake] Add an option to disable source generation

### LINKED ISSUE #6728: Disable the clang-format checks when building DXC
**What is the issue you're encountering?**
When I build the `dxc` target from source as a subdirectory of my codebase, I get these errors:

> 3>Verifying clang-format results...
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(254,5): error MSB8066: Custom build for 'build\win\CMakeFiles\88778fe0101fa821569a6b8d53ded241\RDAT_LibraryTypes.inl.stamp.rule;build\win\CMakeFiles\6f5c683015713da5c9349b7bc8142ac1\RDAT_LibraryTypes.rule' exited with code 1.

A few more targets also fail with this same error. It still produces `dxc.exe` and `dxcompiler.dll`. Is it possible to turn off the clang-format checks? 

**Steps to Reproduce (if applicable)**
1. Clone the latest master
2. Build the dxc target as a subdirectory of another CMake project

**Environment**
- DXC version: `dxcompiler.dll: 1.8 - 1.8.0.2207 (master, 452045c6f-dirty)`
- Host Operating System: Windows 10 22H2 1904.4529
- CMake: `cmake version 3.28.3-msvc11`
- Compiler: `MSVC 19.40.33811.0`

### PR DESCRIPTION
**Pull request by llvm-beanz**
This change adds a new CMake configuration option
`HLSL_DISABLE_SOURCE_GENERATION` which allows a user to disable generating the in-tree sources which contributte to DXC's source releases. This option should only be used by users building DXC and not modifying it.

Resolves #6728

### CODE REVIEW

File: cmake/modules/HCT.cmake [hunk 1]
```diff
@@ -101,7 +107,7 @@ function(add_hlsl_hctgen mode)
   endif()
 
   add_custom_command(OUTPUT ${temp_output}
-                     COMMAND ${Python3_EXECUTABLE}
+                     COMMAND ${PYTHON_EXECUTABLE}
```
Comment (bfavela) [inline cmake/modules/HCT.cmake:110]: This seems unrelated to the intent of the PR? Was it an in-progress change for something local or is it necessary for disabling in-tree sources?

Comment (llvm-beanz) [reply cmake/modules/HCT.cmake]: Doh! I mis-rebased the patch from an out of date branch.

**Review by hekota — APPROVED** [review]

**Review by damyanp — APPROVED** [review]
